src/TesApi.Web/BatchScheduler.cs

```diff
@@ -491,125 +456,171 @@ private static string StandardizeStartTaskPath(string startTaskPath, string defa
         private static bool IsCromwellCommandScript(TesInput inputFile)
             => (inputFile.Name?.Equals("commandScript") ?? false) && (inputFile.Description?.EndsWith(".commandScript") ?? false) && inputFile.Type == TesFileType.FILEEnum && inputFile.Path.EndsWith($"/{CromwellScriptFileName}");
 
-        /// <summary>
-        /// Adds a new Azure Batch pool/job/task for the given <see cref="TesTask"/>
-        /// </summary>
-        /// <param name="tesTask">The <see cref="TesTask"/> to schedule on Azure Batch</param>
-        /// <param name="cancellationToken">A <see cref="CancellationToken"/> for controlling the lifetime of the asynchronous operation.</param>
-        /// <returns>A task to await</returns>
-        private async Task AddBatchTaskAsync(TesTask tesTask, CancellationToken cancellationToken)
+        /// <inheritdoc/>
+        public async IAsyncEnumerable<(TesTask TesTask, Task<bool> IsModifiedAsync)> ProcessQueuedTesTasksAsync(TesTask[] tesTasks, [System.Runtime.CompilerServices.EnumeratorCancellation] CancellationToken cancellationToken)
```

Comment:
Why use a new type instead of multiple return values? A new type can make the method's purpose more clear and allow for more flexibility in adding or removing properties as needed.